Source/UI/VirtusizeInPageView.swift

```diff
@@ -22,97 +22,158 @@
+			self,
+			selector: #selector(didReceiveProductDataCheck(_:)),
+			name: .productDataCheck,
+			object: Virtusize.self
+		)
 
-    internal var loadingTextTimer: Timer?
+		NotificationCenter.default.addObserver(
+			self,
+			selector: #selector(didReceiveStoreProduct(_:)),
+			name: .storeProduct,
+			object: Virtusize.self
+		)
 
-    required init?(coder: NSCoder) {
-        super.init(coder: coder)
-		virtusizeEventHandler = self
-        isHidden = true
-        setup()
-    }
+		NotificationCenter.default.addObserver(
+			self,
+			selector: #selector(didReceiveSizeRecommendationData(_:)),
+			name: .sizeRecommendationData,
+			object: Virtusize.self
+		)
 
-    public override init(frame: CGRect) {
-        super.init(frame: .zero)
-		virtusizeEventHandler = self
-        isHidden = true
-        setup()
-    }
+		NotificationCenter.default.addObserver(
+			self,
+			selector: #selector(didReceiveInPageError(_:)),
+			name: .inPageError,
+			object: Virtusize.self
+		)
+	}
 
-	public override func willMove(toWindow: UIWindow?) {
-		handleWillMoveWindow(toWindow) { isDeallocated in
-			self.isDeallocated = isDeallocated
+	@objc internal func didReceiveProductDataCheck(_ notification: Notification) {
+		shouldUpdateProductDataCheckData(notification) { productWithPDCData in
+			self.clientProduct = productWithPDCData
+			isHidden = false
+			setLoadingScreen(loading: true)
 		}
 	}
 
-	public func isLoading() {
-        isHidden = false
-	}
-
-    internal func setup() {}
-
-    internal func setHorizontalMargins(view: UIView, margin: CGFloat) {
-        view.addConstraint(
-            NSLayoutConstraint(
-                item: self,
-                attribute: .leading,
-                relatedBy: .equal,
-                toItem: view,
-                attribute: .leading,
-                multiplier: 1,
-                constant: margin
-            )
-        )
-        view.addConstraint(
-            NSLayoutConstraint(
-                item: view,
-                attribute: .trailing,
-                relatedBy: .equal,
-                toItem: self,
-                attribute: .trailing,
-                multiplier: 1,
-                constant: margin
-            )
-        )
-    }
-
-    @objc internal func clickInPageViewAction() {
-		openVirtusizeWebView(
-			eventHandler: virtusizeEventHandler
-		)
-    }
+	@objc internal func didReceiveStoreProduct(_ notification: Notification) {
+		shouldUpdateStoreProduct(notification) { storeProduct in
+			self.serverProduct = storeProduct
+		}
+	}
 
 	/// A parent function to set up InPage recommendation
-	internal func setInPageRecommendation(
-		_ externalProductId: String?,
-		_ sizeComparisonRecommendedSize: SizeComparisonRecommendedSize?,
-		_ bodyProfileRecommendedSize: BodyProfileRecommendedSize?
-	) {}
+	@objc internal func didReceiveSizeRecommendationData(_ notification: Notification) {}
+
+	internal func shouldUpdateInPageRecommendation(
```

Comment:
Handle the notification data to determine whether it is the data for this VirtusizeInPage  view (VirtusizeInPageStandard or VirtusizeInPageMini) to update the recommendation